client/src/main/java/org/eclipse/hono/client/impl/AmqpAdapterClientTelemetrySenderImpl.java

```diff
@@ -0,0 +1,123 @@
+
+    AmqpAdapterClientTelemetrySenderImpl(final HonoConnection con, final ProtonSender sender, final String tenantId,
+            final String targetAddress) {
+        super(con, sender, tenantId, targetAddress);
+    }
+
+    /**
+     * Creates a new sender for publishing telemetry data to Hono's AMQP adapter.
+     *
+     * @param con The connection to the Hono server.
+     * @param tenantId The tenant that the telemetry data will be published for.
+     * @param remoteCloseHook The handler to invoke when the link is closed by the peer (may be {@code null}). The
+     *            sender's target address is provided as an argument to the handler.
+     * @return A future indicating the outcome.
+     * @throws NullPointerException if con or tenantId is {@code null}.
+     */
+    public static Future<TelemetrySender> createWithAnonymousLinkAddress(
+            final HonoConnection con,
+            final String tenantId,
+            final Handler<String> remoteCloseHook) {
+
+        Objects.requireNonNull(con);
+        Objects.requireNonNull(tenantId);
+
+        final String targetAddress = null; // anonymous relay
+        return con.createSender(targetAddress, ProtonQoS.AT_LEAST_ONCE, remoteCloseHook)
+                .map(sender -> new AmqpAdapterClientTelemetrySenderImpl(con, sender, tenantId, targetAddress));
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        return send(deviceId, payload, contentType, properties, null);
+    }
+
+    @Override
+    public Future<ProtonDelivery> send(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties, final SpanContext context) {
+        final Message msg = createMessage(deviceId, payload, contentType, properties);
+        return send(msg, context);
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendAndWaitForOutcome(final String deviceId, final byte[] payload,
+            final String contentType, final Map<String, ?> properties) {
+        return sendAndWaitForOutcome(deviceId, payload, contentType, properties, null);
+    }
+
+    @Override
+    public Future<ProtonDelivery> sendAndWaitForOutcome(final String deviceId, final byte[] payload,
+            final String contentType, final Map<String, ?> properties, final SpanContext context) {
+        final Message msg = createMessage(deviceId, payload, contentType, properties);
+        return sendAndWaitForOutcome(msg, context);
+    }
+
+    private Message createMessage(final String deviceId, final byte[] payload, final String contentType,
+            final Map<String, ?> properties) {
+        Objects.requireNonNull(deviceId);
+        Objects.requireNonNull(payload);
+        Objects.requireNonNull(contentType);
+
+        final Message msg = ProtonHelper.message();
```

Comment:
I think we should also set a message id here.

Reply:
This is done further down in `AbstractSender#sendMessageAndWaitForOutcome(Message, Span)`. And for QoS 0 in `TelemetrySenderImpl#sendMessage(Message, Span)`.